internal/plugin.go

```diff
@@ -1068,262 +1062,210 @@ func (p *Plugin) assertValidConfig() error {
-				"policy %s must not specify a placement when generatePlacement is set to false", policy.Name,
+				"each policySet must have a unique name set, but found a duplicate name: %s", plcset.Name,
 			)
 		}
 
-		if policyPlacementOptions > 1 {
-			return fmt.Errorf(
-				"policy %s must specify only one of placement selector, placement path, or placement name", policy.Name,
-			)
-		}
+		seenPlcset[plcset.Name] = true
 
-		if policy.Placement.PlacementRulePath != "" {
-			_, err := os.Stat(policy.Placement.PlacementRulePath)
-			if err != nil {
-				return fmt.Errorf(
-					"could not read the placement rule path %s",
-					policy.Placement.PlacementRulePath,
-				)
-			}
+		// Validate policy set Placement settings
+		err := assertValidPlacement(
+			plcset.Placement, plcset.GeneratePolicySetPlacement, fmt.Sprintf("policySet %s", plcset.Name), &plCount)
+		if err != nil {
+			return err
 		}
+	}
 
-		if policy.Placement.PlacementPath != "" {
-			_, err := os.Stat(policy.Placement.PlacementPath)
-			if err != nil {
-				return fmt.Errorf(
-					"could not read the placement path %s",
-					policy.Placement.PlacementPath,
-				)
-			}
-		}
+	// Validate only one type of placement kind is in use
+	if plCount.plc != 0 && plCount.plr != 0 {
+		return fmt.Errorf(
+			"may not use a mix of Placement and PlacementRule for policies and policysets; found %d Placement and "+
+				"%d PlacementRule",
+			plCount.plc, plCount.plr,
+		)
+	}
 
-		foundPl := false
+	p.usingPlR = plCount.plc == 0
 
-		if len(policy.Placement.LabelSelector) != 0 ||
-			policy.Placement.PlacementPath != "" ||
-			policy.Placement.PlacementName != "" {
-			plCount.plc++
+	return nil
+}
 
-			foundPl = true
-		}
+// assertValidPlacement is a helper for assertValidConfig to verify placement configurations
+func assertValidPlacement(
+	placement types.PlacementConfig,
+	generatePlacement bool,
+	path string,
+	plCount *struct {
+		plc int
+		plr int
+	},
+) error {
+	if placement.PlacementRulePath != "" && placement.PlacementPath != "" {
+		return fmt.Errorf(
+			"%s must provide only one of placement.placementPath or placement.placementRulePath", path,
+		)
+	}
 
-		if len(policy.Placement.ClusterSelectors) != 0 ||
-			policy.Placement.PlacementRulePath != "" ||
-			policy.Placement.PlacementRuleName != "" {
-			plCount.plr++
+	if len(placement.ClusterSelectors) > 0 && len(placement.LabelSelector) > 0 {
+		return fmt.Errorf(
+			"%s must provide only one of placement.labelSelector or placement.clusterSelectors", path,
+		)
+	}
 
-			if foundPl {
-				return fmt.Errorf(
-					"policy %s may not use both Placement and PlacementRule kinds", policy.Name,
-				)
-			}
-		}
+	if placement.PlacementRuleName != "" && placement.PlacementName != "" {
+		return fmt.Errorf(
+			"%s must provide only one of placement.placementName or placement.placementRuleName", path,
+		)
 	}
 
-	seenPlcset := map[string]bool{}
+	if placement.PlacementRulePath != "" && placement.PlacementPath != "" {
```

Comment:
This looks to be a duplicate check from above

Reply:
Ah, indeed--good catch!